frontend/src/metabase/visualizations/echarts/cartesian/model/dataset.ts

```diff
@@ -127,3 +132,43 @@ export const getJoinedCardsDataset = (
 
   return Array.from(groupedData.values());
 };
+
+/**
+ * Sorts the series models based on the order specified in the "graph.series_order" property of visualization settings.
+ * It defines the breakout series order so that charts where all series are metrics without breakouts do not
+ * rely on this property, and the series are sorted based on the metrics order in the original MBQL query.
+ *
+ * @param {SeriesModel[]} seriesModels - The array of series models to be sorted.
+ * @param {ComputedVisualizationSettings} settings - The computed visualization settings.
+ * @returns {SeriesModel[]} The sorted array of series models.
+ */
+export const getSortedSeriesModels = (
+  seriesModels: SeriesModel[],
+  settings: ComputedVisualizationSettings,
+) => {
+  const breakoutSeriesOrder = settings["graph.series_order"];
+  if (breakoutSeriesOrder == null || breakoutSeriesOrder.length === 0) {
+    return seriesModels;
+  }
+
+  const orderedSeriesModels = breakoutSeriesOrder
+    .filter(orderSetting => orderSetting.enabled)
+    .map(orderSetting => {
+      const foundSeries = seriesModels.find(
+        seriesModel => seriesModel.vizSettingsKey === orderSetting.key,
+      );
+      if (foundSeries === undefined) {
+        throw new TypeError("Series not found");
```

Comment:
Nit: I don't think `TypeError` is the best type of error for this, [MDN says `ReferenceError` is for variables that don't exist.]([url](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ReferenceError))